dozer/cogs/actionlogs.py

```diff
@@ -297,27 +320,136 @@ async def messagelogconfig(self, ctx, channel_mentions: discord.TextChannel):
+            embed.add_field(name="Leave template", value=config[0].leave_message, inline=False)
+            embed.add_field(name="Leave Example", value=self.format_join_leave(config[0].leave_message, ctx.author))
+            await ctx.send(embed=embed)
         else:
-            config = GuildMemberLog(guild_id=ctx.guild.id, memberlog_channel=channel_mentions.id, name=ctx.guild.name)
-        await config.update_or_add()
-        await ctx.send(ctx.message.author.mention + ', memberlog settings configured!')
+            await ctx.send("This guild has no member log configured")
 
     memberlogconfig.example_usage = """
-        `{prefix}memberlogconfig #join-leave-logs` - set a channel named #join-leave-logs to log joins/leaves 
+    `{prefix}memberlogconfig setchannel channel`: Sets the member log channel 
+    `{prefix}memberlogconfig toggleping`: Toggles whenever members are pinged upon joining the guild
+    `{prefix}memberlogconfig setjoinmessage template`: Sets join template
+    `{prefix}memberlogconfig setleavemessage template`: Sets leave template
+    `{prefix}memberlogconfig help`: Returns the template formatting key
+    """
+
+    @memberlogconfig.command()
+    @has_permissions(manage_guild=True)
+    async def setchannel(self, ctx, channel: discord.TextChannel):
+        """Configure join/leave channel"""
+        config = CustomJoinLeaveMessages(
+            guild_id=ctx.guild.id,
+            channel_id=channel.id
+        )
+        await config.update_or_add()
+        e = discord.Embed(color=blurple)
+        e.add_field(name='Success!', value=f"Join/Leave log channel has been set to {channel.mention}")
+        e.set_footer(text='Triggered by ' + ctx.author.display_name)
+        await ctx.send(embed=e)
+
+    @memberlogconfig.command()
+    @has_permissions(manage_guild=True)
+    async def toggleping(self, ctx):
+        """Toggles whenever a new member gets pinged on join"""
+        config = await CustomJoinLeaveMessages.get_by(guild_id=ctx.guild.id)
+        if len(config):
+            config[0].ping = not config[0].ping
+        else:
+            config[0] = CustomJoinLeaveMessages(guild_id=ctx.guild.id, ping=True)
+        await config[0].update_or_add()
+
+        e = discord.Embed(color=blurple)
+        e.add_field(name='Success!', value=f"Ping on join is set to: {config[0].ping}")
+        e.set_footer(text='Triggered by ' + ctx.author.display_name)
+        await ctx.send(embed=e)
+
+    @memberlogconfig.command()
+    @has_permissions(manage_guild=True)
+    async def setjoinmessage(self, ctx, *, template: str = None):
+        """Configure custom join message template"""
+        e = discord.Embed(color=blurple)
+        e.set_footer(text='Triggered by ' + ctx.author.display_name)
+        if template:
+            config = CustomJoinLeaveMessages(
+                guild_id=ctx.guild.id,
+                join_message=template
```

Comment:
This (and the same for `setleavemessage`) can likely be simplified using the python `or` operator, so you only have to make the `config` variable once and set `join_message=template or CustomJoinLeaveMessages.nullify` to set it to the appropriate value. This is not a issue that will block merging though.